### ISSUE TITLE
Unknown keras backend NONE

### ISSUE THREAD

**Issue by ryanzpn** [deeplearning4j/deeplearning4j#4395]
I was trying to load keras h5 model by using deeplearning4j-modelimport, but it failed.

I downloaded deeplearning4j master, compiled and installed. The keras version is 2.0.4 and is based on tensorflow.

the error msg is:

```
org.deeplearning4j.nn.modelimport.keras.exceptions.InvalidKerasConfigurationException: Unknown keras backend NONE. For more information, see http://deeplearning4j.org/model-import-keras.
    at org.deeplearning4j.nn.modelimport.keras.layers.convolutional.KerasConvolution.setWeights(KerasConvolution.java:131)
    at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelUtils.importWeights(KerasModelUtils.java:237)
    at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:132)
    at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:60)
    at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:120)
    at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:158)
    at com.hello.world.antispam.learn.KerasToJava.main(KerasToJava.java:14)
```

It seems dimOrder doesn't get parsed from the config

**Comment by agibsonccc** [comment]
@ryanzpn keras 2 support is only on master right now, we will have a release in the next few weeks here.

**Comment by ryanzpn** [comment]
@agibsonccc but I did build it on master

**Comment by maxpumperla** [comment]
@ryanzpn can you please dump out _just_ the model JSON? I've never seen this before and it can only happen if your model doesn't have a `data_format` field in its config, which I'm not sure about why that is.

**Comment by maxpumperla** [comment]
@ryanzpn happy yo look into it once you provide me with what I asked

**Comment by ryanzpn** [comment]
@maxpumperla Sorry, I'm busy with others things these days. The way I use deeplearning4j-modelimport is

```
MultiLayerNetwork model = KerasModelImport.importKerasSequentialModelAndWeights("MODEL_H5_FILE");
```

I thought it would be valid to use in that way, because h5 file contains all the information needed to predict, just like in keras. Do I have to dump a json file?

**Comment by ryanzpn** [comment]
```
{"class_name": "Sequential", "keras_version": "2.0.4", "config": [{"class_name": "Embedding", "config": {"embeddings_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "normal", "scale": 2.0, "seed": null, "mode": "fan_in"}}, "name": "embedding_1", "dtype": "int32", "output_dim": 100, "trainable": true, "embeddings_regularizer": null, "input_dim": 11000, "mask_zero": false, "embeddings_constraint": null, "batch_input_shape": [null, 200], "activity_regularizer": null, "input_length": 200}}, {"class_name": "Conv1D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv1d_1", "bias_regularizer": null, "bias_constraint": null, "activation": "relu", "trainable": true, "padding": "valid", "strides": [1], "dilation_rate": [1], "kernel_regularizer": null, "filters": 1, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3]}}, {"class_name": "MaxPooling1D", "config": {"padding": "valid", "strides": [3], "trainable": true, "name": "max_pooling1d_1", "pool_size": [3]}}, {"class_name": "Conv1D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv1d_2", "bias_regularizer": null, "bias_constraint": null, "activation": "relu", "trainable": true, "padding": "valid", "strides": [1], "dilation_rate": [1], "kernel_regularizer": null, "filters": 1, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3]}}, {"class_name": "MaxPooling1D", "config": {"padding": "valid", "strides": [3], "trainable": true, "name": "max_pooling1d_2", "pool_size": [3]}}, {"class_name": "Flatten", "config": {"trainable": true, "name": "flatten_1"}}, {"class_name": "Dense", "config": {"kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "dense_1", "kernel_constraint": null, "bias_regularizer": null, "bias_constraint": null, "activation": "sigmoid", "trainable": true, "kernel_regularizer": null, "bias_initializer": {"class_name": "Zeros", "config": {}}, "units": 1, "use_bias": true, "activity_regularizer": null}}], "backend": "tensorflow"}
```
@maxpumperla this is the json file generated by calling model.to_json()

**Comment by maxpumperla** [comment]
@ryanzpn no problem, dude. I just wanted to offer help! :)

Of course, we only have one entry point to this application and you're using it correctly. To see what's wrong with your particular use case, I need to see your data. Thanks for the JSON, I'll have a look and let you know what I find.

**Comment by maxpumperla** [comment]
@ryanzpn ok, I found the root cause. I'll try to patch this asap, will keep you posted.

**Comment by maxpumperla** [comment]
@ryanzpn we merged #4418, which should fix this issue. Let me know if you still have problems. thanks

**Comment by ryanzpn** [comment]
[https://pan.baidu.com/s/1qXHFJ44](url)

@maxpumperla Here's the h5 model file, the password is txev

Is it ok to load the h5 file in your environment?

**Comment by maxpumperla** [comment]
@ryanzpn cool, thanks. I'll check this out and put it into our end to end tests.

just fyi, the issue is that in keras 2, if you don't explicitly specify a dim ordering, this is inherited from the backend you choose. before, the dim ordering was properly serialised. this leads to the NONE value you observe.

**Comment by ryanzpn** [comment]
@maxpumperla same problem still exists on master
